### ISSUE TITLE
Support catalyst

### ISSUE THREAD

**Issue by maddanio** [KhronosGroup/MoltenVK#821]
We are planning to convert our Vulkan heavy iOS app to a desktop app using catalyst, and it would be very helpful if mvk would support building a fat framework supporting that

**Comment by billhollings** [comment]
@maddanio 

Would you be able to leverage the knowledge gained in porting your app to add the required changes to MoltenVK and submit a Pull Request?

**Comment by maddanio** [comment]
I will try. So far it seems quite daunting. I had hoped someone else had already made headway :).

**Comment by aerofly** [comment]
We tried to create a Catalyst compatible app also, but we have delayed it a little, until Apple updates their documentation.

What is required for MoltenVK to be compatible with Catalyst is the fact that we need a so called XCFramework to be able to properly link MoltenVK in a Catalyst project.

This can be achieved with a command like this:

`xcodebuild -create-xcframework -library iOS/static/libMoltenVK.a -library macOS/static/libMoltenVK.a -output libMoltenVK.xcframework`

Besides that MoltenVK itself needs to be compiled with a new target, e.g. x86_64-apple-ios13.0-macabi
However, this isn't properly documented yet, so we have put this on hold.

Here is a stack overflow question regarding this issue:
https://stackoverflow.com/questions/56487645/how-to-compile-a-3rd-party-library-to-be-used-with-uikit-for-mac-catalyst
